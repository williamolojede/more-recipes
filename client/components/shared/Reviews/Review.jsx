import React from 'react';
import timeDifference from '../../../utils/timeDifference';
import UserImg from '../UserImg.jsx';
import { reviewPropTypes } from '../../../config/proptypes';

const Review = ({ review }) => (
  <li className="review__card">
    <div className="wrapper">
      <UserImg user={review.User} type="inReview" />
      <div className="review__card-card-content">
        <div className="review__card-card-content___top">
          <p className="user-name">{review.User.fullname}</p>
          <span className="time-since">{timeDifference(review.createdAt)}</span>
        </div>
        <p className="review__card-card-content">{review.content}</p>
      </div>
    </div>
  </li>
);

Review.propTypes = {
  ...reviewPropTypes
};
export default Review;
