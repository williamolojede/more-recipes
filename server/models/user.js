import bcrypt from 'bcrypt';

module.exports = (sequelize, DataTypes) => {
  const User = sequelize.define('User', {
    username: {
      allowNull: true,
      unique: true,
      validate: {
        max: 15,
      },
      type: DataTypes.STRING,
    },
    email: {
      allowNull: false,
      unique: true,
      validate: {
        isEmail: true,
      },
      type: DataTypes.STRING,
    },
    fullname: {
      allowNull: false,
      validate: {
        max: 50,
      },
      type: DataTypes.STRING,
    },
    password: {
      allowNull: false,
      type: DataTypes.STRING,
      set(password) {
        const hash = bcrypt.hashSync(password, 10);
        this.setDataValue('password', hash);
      }
    },
  });

  User.associate = (models) => {
    User.hasMany(models.Recipe, {
      foreignKey: 'owner',
      as: 'recipes',
    });

    User.hasMany(models.Favorite, {
      foreignKey: 'userId',
      as: 'favorites',
    });

    User.hasMany(models.Review, {
      foreignKey: 'userId',
      as: 'reviews',
    });
  };

  User.authenticate = (email, password, callback) => {
    // console.log(email, password);
    User.findOne({
      where: { email },
    })
      .then((user) => {
        // check if it finds no user
        if (!user) {
          const err = new Error('User not found');
          err.status = 404;
          return callback(err);
        }
        const isMatch = bcrypt.compareSync(password, user.password);

        // check if password matches
        if (isMatch) return callback(null, user);
        return callback();
      })
      .catch(err => callback(err));
  };
  return User;
};
