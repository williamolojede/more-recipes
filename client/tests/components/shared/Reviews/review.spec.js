import React from 'react';
import Review from '../../../../components/shared/Reviews/Review';
import user from '../../../__mocks__/user.mock';

function setup() {
  const props = {
    review: {
      content: 'Shitty recipe',
      User: user
    }
  };
  const component = shallow(<Review {...props} />);
  return {
    component
  };
}


describe('Notification', () => {
  it('should render with right amount of elements', () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });
});
