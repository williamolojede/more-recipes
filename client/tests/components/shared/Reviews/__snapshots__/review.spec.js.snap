// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should render with right amount of elements 1`] = `
<li
  className="review__card"
>
  <div
    className="wrapper"
  >
    <UserImg
      type="inReview"
      user={
        Object {
          "fullname": "John Doe",
          "id": 1,
        }
      }
    />
    <div
      className="review__card-card-content"
    >
      <div
        className="review__card-card-content___top"
      >
        <p
          className="user-name"
        >
          John Doe
        </p>
        <span
          className="time-since"
        >
          a few seconds ago
        </span>
      </div>
      <p
        className="review__card-card-content"
      >
        Shitty recipe
      </p>
    </div>
  </div>
</li>
`;
